compiler/ETA/CodeGen/Constr.hs

```diff
@@ -41,18 +41,24 @@ cgTopRhsCon dflags id dataCon args = (cgIdInfo, genCode)
         -- typeFt = obj (tyConClass dflags (dataConTyCon dataCon))
         genCode = do
           loads <- mapM getArgLoadCode . getNonVoids $ zip maybeFields args
-          deps <- getArgReferences . getNonVoids $ zip maybeFields args
-          defineField $ mkFieldDef [Public, Static] qClName closureType
-          let field = (mkFieldRef modClass qClName closureType)
-          addInitStep (
-              new dataFt
-           <> dup dataFt
-           <> fold loads
-           <> invokespecial (mkMethodRef dataClass "<init>" fields void)
-           <> putstatic field
-           , field
-           , deps
-           )
+          defineField $ mkFieldDef [Private, Static] qClName closureType
+          let field = mkFieldRef modClass qClName closureType
+              loadCodes =
+                [
+                  new dataFt
+                , dup dataFt
+                , fold loads
+                , invokespecial (mkMethodRef dataClass "<init>" fields void)
+                , putstatic field
+                ]
+          defineMethod . mkMethodDef' [Public, Static] qClName (mkMethodDesc [] (Just closureType)) $ fold
+            [
+              getstatic field
+            , markStackMap
```

Comment:
I guess I should mark where the jump actually is.

```
Exception in thread "main" java.lang.VerifyError: Expecting a stackmap frame at branch target 19
Exception Details:
  Location:
    main/Main.DZCmain_closure()Leta/runtime/stg/StgClosure; @3: ifnonnull
  Reason:
    Expected stackmap frame at this location.
  Bytecode:
    0x0000000: b200 11c7 0010 bb00 0559 b700 16b3 0011
    0x0000010: a700 03b2 0011 b0                      

        at eta.main.main(Unknown Source)
```

But why does this fix the above?